### ISSUE TITLE
Provide `wheel` distribution

### ISSUE THREAD

**Issue by spooning** [robotframework/robotframework#1734]
> _Originally submitted to [Google Code](http://code.google.com/p/robotframework/issues/detail?id=1734) by @pekkaklarck on 17 Jun 2014_

RF 2.8.5 added possibility to create wheels (http://wheel.rtfd.org/, #1674). Unfortunately the wheel we initially uploaded to PyPI broke installation with pip on Windows and needed to be removed. The reason was that our setup.py controls which start-up scripts to create and with wheels setup.py is not executed.

In RF 2.9 we should take a new look at this. Most likely it requires making changes to how start-up scripts are created. An important task is also documenting wheel support.

**Comment by RonnyPfannschmidt** [comment]
as far as i can tell, all you would have to do is to add entrypoints for the script names
it would properly work instead of falling over like it does now

what are the big changes needed?

**Comment by pekkaklarck** [comment]
We currently create `pybot` script for Python, `jybot` for Jython and `ipybot` for IronPython. I don't think that's possible with wheels. Creating different start-up script for different interpreters may not be that good idea in general, but that's not something we want to change in 2.9.

**Comment by RonnyPfannschmidt** [comment]
python ironpython and jython have different abi tags, so it seems rather possible to just make wheels with such implementation, however its a bit more of a mess to make those wheels

its indeed uncertain if such a move is a good idea

**Comment by pekkaklarck** [comment]
I think that is a good idea. But it definitely isn't a good idea right now when we want to get 2.9 released in few weeks.

**Comment by RonnyPfannschmidt** [comment]
would it be an acceptable solution to publish such wheels as part of the 2.9 series?

**Comment by pekkaklarck** [comment]
The problem is that if we put a wheel to PyPI, pip will automatically use it. With our custom `setup.py` that causes broken installations at least on Windows (we've done that). We obviously could put wheels somewhere else, e.g. include them into the release on GitHub, but I doubt anyone will use them. If someone actually wants a wheel, it's always possible to clone the repository, checkout the right tag, and create a wheel with `python setup.py bdist_wheel`.

**Comment by pekkaklarck** [comment]
The problem with start-up scripts is getting easier now that RF 3.0 introduced new `robot` start-up script (#2216). I don't think we can remove the old scripts in RF 3.1 yet, though.

**Comment by pekkaklarck** [comment]
Tried to add a zip distro in addition to tar.gz to ease manual installation on Windows (#2542), but it turned out that it's possible to upload only one sdist to PyPI nowadays. If we get wheels working, could consider should we change the sdist format to zip altogether.

**Comment by mblakley** [comment]
I recently tried running "pip wheel" to generate a local wheel file from my virtual environment (using robotframework 3.0.2), and when I installed the wheel file on a different PC, the robot.bat file contained the full python.exe path from my virtual environment.  ex: C:\home\automation\env\Scripts\python.exe -m robot.run %* 

I found this out because robot.bat wouldn't run, because the other machine didn't have python in the same path.  You probably don't want to use the full path - just assume that python is already on the path.

**Comment by pekkaklarck** [comment]
@mblakley, we actually used absolute paths in `*.bat`hg files on purpose. If we had just used `python` and you had, for example, `C:\Python27`, in PATH, running, for example, `C:\Python36\Scripts\robot.bat` would execute Python 2.7. We could have tried handling that with special variables like `%~dp0`, but that would have still left some corner cases. As @RonnyPfannschmidt commented, there's a reason why setuptools entry-points create these `*.exe` wrappers. After #2415 fixed we also use this approach so the problem ought to be finally solved.
